src/components/Card.js

```diff
@@ -1,32 +1,69 @@
 class Card {
-  constructor(data, cardSelector, handleCardClick) {
+  constructor(
+    data,
+    cardSelector,
+    handleCardClick,
+    handleDeleteClick,
+    handleLikeClick,
+    currentUserId
+  ) {
     this._name = data.name;
     this._link = data.link;
+    this._owner = data.owner;
     this._cardSelector = cardSelector;
     this._handleCardClick = handleCardClick;
+    this._handleDelete = handleDeleteClick;
+    this._handleLike = handleLikeClick;
+    this._likes = Array.isArray(data.likes) ? data.likes : [];
+    this._cardID = data._id;
+    this._isLiked = data.isLiked;
+    this._currentUserId = currentUserId;
+  }
+
+  isLiked() {
+    return this._likes.some((like) => like._id === this._owner._id);
+  }
+
+  likeCountAdd() {
+    this._handleLike(this);
+  }
+
+  likeCountRemove() {
+    this._handleLikeRemove(this._cardID);
   }
 
   _setEventListeners() {
     const likeButton = this._element.querySelector(".card__like-button");
     const deleteButton = this._element.querySelector(".card__delete-button");
     const cardImage = this._element.querySelector(".card__image");
 
-    likeButton.addEventListener("click", () => this._handleLikeIcon());
-
-    deleteButton.addEventListener("click", this._handleDelete.bind(this));
+    likeButton.addEventListener("click", () => this._handleLike(this));
+    debugger;
+    if (this._owner._id === this._currentUserId) {
+      deleteButton.style.display = "block"; // Show the trash button
+      deleteButton.addEventListener("click", () =>
+        this._handleDelete(this, this._cardID)
+      );
+    } else {
+      deleteButton.style.display = "none"; // Hide the trash button
```

Comment:
Please remove the trash button from the document instead of adding a class to it, so the user won't be able to click on the button even using the console `deleteButton.remove()`